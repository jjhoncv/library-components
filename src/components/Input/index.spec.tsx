import * as React from 'react';
import { shallow } from 'enzyme';
import { Input } from './index';

let propsInput = {
  className:'c-input',
  type:'text',
  name:'input',
  width:'290px',
  placeholder:'Ingrese el texto aquÃ­',
  value:''
}

describe('Input Component', () => {
  describe('Input => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Input {...propsInput} />
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Input Builder', () => {
    it('Input => render with name', () => {
      const props = {
        name: 'test'
      }
      const wrapper = shallow(
        <Input {...props} />
      )
      expect(wrapper.childAt(0).prop('name')).toEqual(props.name);
    })
    it('Input => render with value', () => {
      const props = {
        name: 'test-input',
        value: 'Ingresa texto...',
        onChange: () => {},
      }
      const wrapper = shallow(
        <Input {...props} />
      );
      expect(wrapper.childAt(0).prop('defaultValue')).toEqual(props.value);
    });
  });

});


