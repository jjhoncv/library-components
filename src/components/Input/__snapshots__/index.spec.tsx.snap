// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Input => render snapshot Render correctly 1`] = `"<div class=\\"sc-bdVaJa lntvKF\\"><input type=\\"text\\" class=\\"c-input sc-bwzfXH bvEjUc\\" name=\\"input\\" placeholder=\\"Ingrese el texto aquÃ­\\" width=\\"290px\\" value=\\"\\"/></div>"`;
